### PR TITLE
Feature/data base added

### PR DESCRIPTION
**Pull request by ZagreDo**
Created new (REST) API,  for interactions with Database (MSSQL) and back-end logic. Db is created with EF core. Client Encounter_Me project access API through HttpClient Dependency Injection. There are 2 tables in database : User and Trail. Such functionality as adding, deleting elements, getting details exists.

### CODE REVIEW

File: Encounter Me.Api/Controllers/TrailController.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                return BadRequest();
+
+            var trailToDelete = _trailRepository.GetTrailById(id);
+            if (trailToDelete == null)
+                return NotFound();
+
+            _trailRepository.DeleteTrail(id);
+
+            return NoContent();//success
+        }
```
Comment (GabijaG) [inline Encounter Me.Api/Controllers/TrailController.cs:59]: I suggest commenting these methods as for now we do not have any use for them - there is no UI to modify trails on the client. There is nothing on back-end side as well. Also, why NoContent is a successful outcome?

File: Encounter Me.Api/Controllers/UserController.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                return BadRequest();
+
+            var userToDelete = _userRepository.GetUserById(id);
+            if (userToDelete == null)
+                return NotFound();
+
+            _userRepository.DeleteUser(id);
+
+            return NoContent();//success
+        }
```
Comment (GabijaG) [inline Encounter Me.Api/Controllers/UserController.cs:91]: Here the same as with TrailController - these methods will be useful if we will do CRUD for UI, but not now.

Comment (ZagreDo) [reply Encounter Me.Api/Controllers/UserController.cs:91]: I'm waiting you to fix it, as in agreement.

**Review by GabijaG — APPROVED** [review]

**Review by GabijaG — APPROVED** [review]
I'll take care of those methods in Trail controller in my own brach jwt authentication.

**Pull request merged**